### PR TITLE
ffmpeg_4: 4.4.4 -> 4.4.5

### PR DESCRIPTION
**Pull request by jopejoe1**
## Description of changes

ChangeLog: https://github.com/FFmpeg/FFmpeg/blob/n4.4.5/Changelog

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by Atemu — APPROVED** [review]
Diff LGTM but I'd like to see this built on a Darwin machine before we merge.

**Comment by emilazy** [comment]
Can you rebase the revision on `master` so I can test the build on Darwin more easily? (While keeping the branch target as it is, of course.)

If I just got the changes I have locally pushed out this wouldn’t need to target `staging` at all…

**Comment by jopejoe1** [comment]
> Can you rebase the revision on `master` so I can test the build on Darwin more easily? (While keeping the branch target as it is, of course.)

sure

> If I just got the changes I have locally pushed out this wouldn’t need to target `staging` at all…

Would be nice to not have to to through staging.

File: pkgs/development/libraries/ffmpeg/generic.nix [hunk 1]
```diff
@@ -403,36 +403,6 @@ stdenv.mkDerivation (finalAttrs: {
...
-        url = "https://git.ffmpeg.org/gitweb/ffmpeg.git/patch/031f1561cd286596cdb374da32f8aa816ce3b135";
-        hash = "sha256-agJgzIzrBTQBAypuCmGXXFo7vw6Iodw5Ny5O5QCKCn8=";
-      })
-      (fetchpatch2 {
-        # Backport fix for binutils-2.41.
-        name = "binutils-2.41.patch";
-        url = "https://git.ffmpeg.org/gitweb/ffmpeg.git/patch/effadce6c756247ea8bae32dc13bb3e6f464f0eb";
-        hash = "sha256-vLSltvZVMcQ0CnkU0A29x6fJSywE8/aU+Mp9os8DZYY=";
-      })
-      # The upstream patch isn’t for ffmpeg 4, but it will apply with a few tweaks.
```
Comment (emilazy) [inline pkgs/development/libraries/ffmpeg/generic.nix:418]: Just confirming: I guess they did backport it to FFmpeg 4 eventually?

**Review by emilazy — APPROVED** [review]
> Would be nice to not have to to through staging.

[Slowly](https://github.com/NixOS/nixpkgs/pull/327642) [but](https://github.com/NixOS/nixpkgs/pull/329470) [surely](https://github.com/NixOS/nixpkgs/pull/330648), [I’m](https://github.com/NixOS/nixpkgs/pull/331522) [making](https://github.com/NixOS/nixpkgs/pull/331558) [it](https://github.com/NixOS/nixpkgs/pull/331568) [happen](https://github.com/NixOS/nixpkgs/pull/331615).

I can confirm a successful build of all variants on both `x86_64-darwin` and `aarch64-darwin`. Hey, if we’re lucky maybe this will be the last time we have to bump FFmpeg 4…

File: pkgs/development/libraries/ffmpeg/generic.nix (hunk 1 above)
Comment (Atemu) [reply pkgs/development/libraries/ffmpeg/generic.nix:418]: In the changelog you can find the commit title: https://github.com/FFmpeg/FFmpeg/blob/9bcede27c26b2f7cd469ab6b5c8b9694c30cfca3/Changelog#L218

**Comment by Atemu** [comment]
Do we really need this backported? Don't we have all the security fixes ported as patches already?

**Pull request merged**